libpromises/evalfunction.c

```diff
@@ -7909,17 +7909,52 @@ static bool ClassFilterDataArray(
 }
 
 static bool ClassFilterDataObjectOfArrays(
-    ARG_UNUSED EvalContext *ctx,
-    ARG_UNUSED const char *fn_name,
-    ARG_UNUSED JsonElement *json_array,
-    ARG_UNUSED const char *class_expr_index,
-    ARG_UNUSED const char *exogenous_key,
-    ARG_UNUSED bool *remove)
-{
-    /* Should already have been caught by syntax checker, since it is not yet
-     * added to the validation pattern in CLASSFILTERDATA_ARGS. */
-    ProgrammingError("classfilterdata() with data structure object of arrays is not yet implemented");
-    return false;
+    EvalContext *ctx,
+    const char *fn_name,
+    JsonElement *json_array,
+    const char *class_expr_index,
+    const char *exogenous_key,
+    bool *remove)
+{
+    /* Use the key of the object itself as the class expression, unless a class
+     * expression index is specified */
+    const char *class_expr = exogenous_key;
+    if (class_expr_index != NULL)
+    {
+        size_t index;
+        assert(SIZE_MAX >= ULONG_MAX); /* make sure returned value can fit in size_t */
+        if (StringToUlong(class_expr_index, &index) != 0) {
+            Log(LOG_LEVEL_VERBOSE,
+                "Function %s(): Bad class expression index '%s': Failed to parse integer",
```

Comment:
is an index a class expression? shouldn't that just be an index, aka integer and specific to this function as an argument.